lib/carbonite/query.ex

```diff
@@ -76,31 +77,107 @@ defmodule Carbonite.Query do
   @spec current_transaction([current_transaction_option()]) :: Ecto.Query.t()
   def current_transaction(opts \\ []) do
-    carbonite_prefix = get_carbonite_prefix(opts)
-
     from(t in Transaction, where: t.id == fragment("pg_current_xact_id()"))
-    |> put_query_prefix(carbonite_prefix)
     |> maybe_preload(opts, :changes)
+    |> put_carbonite_prefix(opts)
+  end
+
+  @doc """
+  Returns an `t:Ecto.Query.t/0` that selects a outbox by name.
+
+  ## Options
+
+  * `carbonite_prefix` - defines the audit trail's schema, defaults to `"carbonite_default"`
+  """
+  @doc since: "0.4.0"
+  @spec outbox(Outbox.name()) :: Ecto.Query.t()
+  @spec outbox(Outbox.name(), [prefix_option()]) :: Ecto.Query.t()
+  def outbox(outbox_name, opts \\ []) do
+    from(o in Outbox)
+    |> where([o], o.name == ^outbox_name)
+    |> put_carbonite_prefix(opts)
+  end
+
+  @type outbox_queue_option ::
+          prefix_option()
+          | preload_option()
+          | {:min_age, non_neg_integer() | disabled()}
+          | {:batch_size, non_neg_integer() | disabled()}
+
+  @doc """
+  Returns an `t:Ecto.Query.t/0` that selects the next batch of transactions for an outbox.
+
+  * Transactions are ordered by their ID ascending, so *roughly* in order of insertion.
+
+  ## Options
+
+  * `min_age` - the minimum age of a record, defaults to 300 seconds (set nil to disable)
+  * `batch_size` - limits the query in size, defaults to 100 (set nil to disable)
+  * `carbonite_prefix` - defines the audit trail's schema, defaults to `"carbonite_default"`
+  * `preload` - can be used to preload the changes, defaults to `true`
+  """
+  @doc since: "0.4.0"
+  @spec outbox_queue(Outbox.t()) :: Ecto.Query.t()
+  @spec outbox_queue(Outbox.t(), [outbox_queue_option()]) :: Ecto.Query.t()
+  def outbox_queue(%Outbox{last_transaction_id: last_processed_tx_id}, opts \\ []) do
+    opts =
+      opts
+      |> Keyword.put_new(:preload, true)
+
+    from(t in Transaction)
+    |> where([t], t.id > ^last_processed_tx_id)
+    |> maybe_apply(opts, :batch_size, 100, fn q, bs -> limit(q, ^bs) end)
+    |> maybe_apply(opts, :min_age, 300, &where_inserted_at_lt/2)
+    |> maybe_preload(opts, :changes)
+    |> order_by({:asc, :id})
+    |> put_carbonite_prefix(opts)
+  end
+
+  @type outbox_done_option :: prefix_option() | {:min_age, non_neg_integer() | disabled()}
+
+  @doc """
+  Returns an `t:Ecto.Query.t/0` that selects all completely processed transactions.
+
+  * If no outbox exists, this query returns all transactions.
+  * If one or more outboxes exist, this query returns all transactions with an ID less than the
+    minimum of the `last_transaction_id` attributes of the outboxes.
+  * Transactions are not ordered.
+
+  ## Options
+
+  * `min_age` - the minimum age of a record, defaults to 300 seconds (set nil to disable)
+  * `carbonite_prefix` - defines the audit trail's schema, defaults to `"carbonite_default"`
+  * `preload` - can be used to preload the changes, defaults to `true`
```

Comment:
apparently we can also preload with `atoms` maybe this should be documented.

Reply:
It is sort of document by the typespecs, IMO this should be enough for now. Trying not to overwhelm the docs with details so much now, especially for "features" we hadn't ourselves yet have the use for (like passing anything but `true` into the `preload` option here)